vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java

```diff
@@ -574,16 +573,17 @@ public void removeThemeVariants(NotificationVariant... variants) {
                         .collect(Collectors.toList()));
     }
 
-    private void attachComponentTemplate() {
-        deferredJob = new AttachComponentTemplate();
-        getElement().getNode().runWhenAttached(ui -> ui
-                .beforeClientResponse(this, context -> deferredJob.accept(ui)));
+    private void configureComponentRenderer() {
+        configureTemplate = new ConfigureComponentRenderer();
+        getElement().getNode()
+                .runWhenAttached(ui -> configureTemplate.accept(ui));
```

Comment:
This call is still required, the component might already be attached when adding components to it.

Also it seems there is no need to defer to before client response.